### ISSUE TITLE
WebChimera.js Linux support/fallback

### ISSUE THREAD

**Issue by thangngoc89** [amilajack/popcorn-time-desktop#49]
I'm having a hard time setting the dev environment on both Windows and ElementaryOS.

The install stucks at this step:

```
> wcjs-prebuilt@3.0.0 postinstall /home/khoa/web/popcorn/node_modules/wcjs-prebuilt
> node install.js

Fetching WebChimera prebuilt for Electron: 
WebChimera version: v0.2.6 
Electron version: v1.2.6 
Platform: linux 
Arch: x64 
Target dir: ./bin

Looking for WebChimera download at https://api.github.com/repos/RSATom/WebChimera.js/releases/tags/v0.2.6

WebChimera.js_electron_0.33.9_ia32_win.zip  doesn't include VLC 
WebChimera.js_electron_0.37.6_ia32_win.zip  doesn't include VLC 
WebChimera.js_electron_0.37.6_x64_osx.zip  doesn't include VLC 
WebChimera.js_electron_0.37.8_ia32_win.zip  doesn't include VLC 
WebChimera.js_electron_0.37.8_x64_osx.zip  doesn't include VLC 
WebChimera.js_electron_1.2.6_ia32_win.zip  doesn't include VLC 
WebChimera.js_electron_1.2.6_x64_osx.zip  doesn't include VLC 
WebChimera.js_electron_1.2.6_x64_win.zip  doesn't include VLC 
WebChimera.js_nw_0.15.4_ia32_win.zip  doesn't include VLC 
WebChimera.js_nw_0.15.4_x64_osx.zip  doesn't include VLC 
WebChimera.js_nw_0.15.4_x64_win.zip  doesn't include VLC 
WebChimera.js_v0.2.6_electron_v0.33.9_VLC_v2.2.4_ia32_win.zip  not matching environment 
WebChimera.js_v0.2.6_electron_v0.37.6_VLC_v2.2.4_ia32_win.zip  not matching environment 
WebChimera.js_v0.2.6_electron_v0.37.6_VLC_v2.2.4_x64_osx.tar.gz  not matching environment 
WebChimera.js_v0.2.6_electron_v0.37.8_VLC_v2.2.4_ia32_win.zip  not matching environment 
WebChimera.js_v0.2.6_electron_v0.37.8_VLC_v2.2.4_x64_osx.tar.gz  not matching environment 
WebChimera.js_v0.2.6_electron_v1.2.6_VLC_v2.2.4_ia32_win.zip  not matching environment 
WebChimera.js_v0.2.6_electron_v1.2.6_VLC_v2.2.4_x64_osx.tar.gz  not matching environment 
WebChimera.js_v0.2.6_electron_v1.2.6_VLC_v2.2.4_x64_win.zip  not matching environment 
WebChimera.js_v0.2.6_nw_v0.15.4_VLC_v2.2.4_ia32_win.zip  not matching environment 
WebChimera.js_v0.2.6_nw_v0.15.4_VLC_v2.2.4_x64_osx.tar.gz  not matching environment 
WebChimera.js_v0.2.6_nw_v0.15.4_VLC_v2.2.4_x64_win.zip  not matching environment 

No WebChimera release found matching your environment
```

Did you setup an dev environment on ElementaryOS ? Any advice?

**Comment by amilajack** [comment]
Oh right. WebChimera.js does not support Linux at the moment :(

https://github.com/RSATom/WebChimera.js/issues/69

**Comment by amilajack** [comment]
Does it work on Windows?

**Comment by thangngoc89** [comment]
@amilajack on Windows it stucks at build step for node-gyp. I can't find a workaround for it. And developing node app on Windows is terribly slow.

**Comment by amilajack** [comment]
Hmm WebChimera.js is an optional dependency. Does it throw an error and quit the npm entire installation?

**Comment by thangngoc89** [comment]
I checked out release-0.0.3-alpha branch. Now it's working

**Comment by amilajack** [comment]
Oh right. The dev-master and master haven't been merged yet. Waiting for ci to finish before i merge

**Comment by amilajack** [comment]
Actually I think it would be best to reopen this issue and rename it to 'WebChimera.js Linux support/fallback'

**Comment by thangngoc89** [comment]
Sounds good to me

**Comment by amilajack** [comment]
I am not sure how butter-desktop solves this problem. They aren't using WebChimera.js. Here are the possible solutions that I have found:

1. @jayke-vbh 's 'dirty hack': https://github.com/RSATom/WebChimera.js/issues/69#issuecomment-233915660

2. Fallback to VLC player. This is is a less elegant solution because the user has to leave the app and the user is required to have VLC. Good thing is that VLC is running natively, so performance should be optimal. 

3. Fallback to https://github.com/duanyao/codecbox.js. The issue with this is that I feel that the project is inactive and it doesnt provide bindings. Its also pretty slow

**Comment by thangngoc89** [comment]
No idea how but PowerPlayer supports Linux. See https://github.com/jaruba/PowderPlayer/releases

**Comment by amilajack** [comment]
Its using an older version of electron (0.37.6)

**Comment by amilajack** [comment]
@thangngoc89 added some initial support for this in the 'torrent wins'.

Make sure you have vlc player installed first.

```bash
git pull --all
git checkout torrent-wins
npm i && npm update
npm run dev
```

This should work for any tv show or torrent

**Comment by thangngoc89** [comment]
Ollie. Sounds good

**Comment by thangngoc89** [comment]
So you'll go for a custom build of Electron right?

**Comment by amilajack** [comment]
Its something similar. I'll replace some files during installation.

**Comment by amilajack** [comment]
Fixed in release 0.0.5
